tasks/sign-index-image/sign-index-image.yaml

```diff
@@ -46,48 +49,69 @@ spec:
         set -e
 
+        TASK_LABEL="internal-services.appstudio.openshift.io/group-id"
+        TASK_ID=$(context.taskRun.uid)
+        PIPELINERUN_LABEL="internal-services.appstudio.openshift.io/pipelinerun-uid"
+
         DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
         if [ ! -f "${DATA_FILE}" ] ; then
             echo "No valid data file was provided."
             exit 1
         fi
 
+        requestType=$(jq -r '.sign.requestType // "internal-request"' "${DATA_FILE}")
+        if [ "${requestType}" == "internal-pipelinerun" ] ; then
+          RPA_FILE="$(workspaces.data.path)/$(params.releasePlanAdmissionPath)"
+          if [ ! -f "${RPA_FILE}" ] ; then
+              echo "No valid rpa file was provided."
+              exit 1
+          fi
+          service_account_name=$(jq -r '.spec.pipeline.serviceAccountName // "appstudio-pipeline"' "${RPA_FILE}")
+          EXTRA_ARGS=(
+          --task-git-url "$(params.taskGitUrl)"
+          --task-git-revision "$(params.taskGitRevision)"
+          --service-account "${service_account_name}"
+          )
+        else
+          requestType=internal-request
+          EXTRA_ARGS=()
+        fi
         request=$(jq -r '.sign.request // "simple-signing-pipeline"' "${DATA_FILE}")
+
         default_pipeline_image="quay.io/redhat-isv/operator-pipelines-images:released"
         pipeline_image=$(jq -r --arg default_pipeline_image "${default_pipeline_image}" \
             '.sign.pipelineImage // .fbc.pipelineImage // $default_pipeline_image' "${DATA_FILE}")
         config_map_name=$(jq -r '.sign.configMapName // .fbc.configMapName // "signing-config-map"' "${DATA_FILE}")
         reference_image="$(params.referenceImage)"
 
-        pipelinerun_label="internal-services.appstudio.openshift.io/pipelinerun-uid"
-
         # Translate direct quay.io reference to public facing registry reference
         # quay.io/redhat/product----repo -> registry.redhat.io/product/repo
         reference_image=$(translate-delivery-repo "$reference_image" | jq -r '.[] | select(.repo=="redhat.io") | .url')
 
         # get all digests from manifest list
         for manifest_digest in $(params.manifestListDigests)
         do
+
           echo "Creating InternalRequest to sign image:"
           echo "- reference=${reference_image}"
           echo "- manifest_digest=${manifest_digest}"
           echo "- requester=$(params.requester)"
+  
+          ${requestType} -r "${request}" \
+            -p pipeline_image="${pipeline_image}" \
+            -p reference="${reference_image}" \
+            -p manifest_digest="${manifest_digest}" \
+            -p requester="$(params.requester)" \
+            -p config_map_name="${config_map_name}" \
+            -l ${TASK_LABEL}="${TASK_ID}" \
+            -l ${PIPELINERUN_LABEL}="$(params.pipelineRunUid)" \
+            -t "$(params.requestTimeout)" --pipeline-timeout "0h30m0s" --task-timeout "0h25m0s" \
+            "${EXTRA_ARGS[@]}" -s false
```

Comment:
With `-s false` now, won't we exit as soon as we kick off each one? There doesn't seem to be anything waiting for them to finish

Reply:
wow. nice catch! fixed!